### PR TITLE
DHFPROD-8018: Basic TABLE component implementation

### PR DESCRIPTION
**Pull request by mrodrigu-marklogic**
### Description

Replaces `antd` `<Table />` component in favor of a new wrapper called `<HCTable />. This PR adds the new component and 
replaces the instances on the Load and Modeling views.

**Note: screenshots subject to further updates**
**Load**
image

**Modeling** (1/2)
image

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
- [x] Run UI tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Comment by mrodrigu-marklogic** [comment]
Currently, 8037 (pagination API), 8038 (column sorting) and 8043 (expand + collapse rows) are ready for review, and 8041 (customizing styles for the replaced tables from 8018) is getting wrapped up, so getting an early review for those implementations would be useful, thank you!

File: marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx [hunk 1]
```diff
@@ -131,7 +131,33 @@ const EntityMapTable: React.FC<Props> = (props) => {
     </ul></div>
   </div>;
 
-  const getColumnsForEntityTable: any = () => {
+  const getColumnsForEntityTable: any = (newTable?: boolean) => {
+    if (newTable) {
+      return entityColumns.map(el => props.checkedEntityColumns[el.key] ? el : "").filter(item => item).map(el => {
+        //console.log("ELEMENT", el);
```
Comment (federicoRufrancosMarklogic) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx:137]: all this comment out lines are needed? Othewise clean up

Comment (mrodrigu-marklogic) [reply marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx]: oops, sandbox code to be added on DHFPROD-7747, good catch!

**Review by briantang — APPROVED** [review]
Works well now! Couple style issues will be split off into a separate ticket. This may have a couple test failures - will monitor it.

**Review by Sanjeevani19 — COMMENTED** [review]
@mrodrigu-marklogic Looks good, any reason for not adding a RTL test to table component ?

**Comment by mrodrigu-marklogic** [comment]
Originally I intended to write a test for the HCTable but I was afraid to miss some of the cases this PR didn't cover, so went on to check using the tests. Does it make sense to evolve it incrementally? @Sanjeevani19

**Comment by Sanjeevani19** [comment]
> Originally I intended to write a test for the HCTable but I was afraid to miss some of the cases this PR didn't cover, so went on to check using the tests. Does it make sense to evolve it incrementally? @Sanjeevani19

@mrodrigu-marklogic Makes sense, Can we write at-least the basic tests that the table, pagination, sorting etc are rendering properly

**Comment by briantang** [comment]
Just a thought, but I know the components using the table itself already had some decent table testing created during story development. So thought coverage with that was okay for now.

However, I have noticed that the ant replacement team has been adding tests specific to the replaced component. Maybe there was a reason for this approach. I could see this being redundant sometimes but also some basic coverage would not hurt.

In this case, I would lean towards just getting this PR through for now as it trailblazes some of the table replacement work. I think we can rely on the test coverage that was implemented during story development for the components utilizing the tables. For example, I see load-list.tsx already has RTL tests for sorting and pagination. As long as we make sure those tests still pass, we can probably proceed? What do you think, @Sanjeevani19  @mrodrigu-marklogic ?

**Comment by mrodrigu-marklogic** [comment]
All good @Sanjeevani19 @briantang, I added a basic RTL test for pagination, sorting and no data checks for the new table component.

**Review by Sanjeevani19 — DISMISSED** [review]

**Comment by briantang** [comment]
Reviewed some test failures with @mrodrigu-marklogic . There is some work to be done there. Everything else looks good. Converting to draft for now.

**Review by briantang — DISMISSED** [review]
All tests passed on latest run. Looks good now!

**Review by Sanjeevani19 — DISMISSED** [review]

**Review by Sanjeevani19 — DISMISSED** [review]

**Review by briantang — APPROVED** [review]

**Review by Sanjeevani19 — APPROVED** [review]

**Pull request merged**